tests/data/annotation_import/test_generic_data_types.py

```diff
@@ -0,0 +1,336 @@
+    [
+        "audio",
+        lb_types.AudioData,
+        [radio_annotation, checklist_annotation, text_annotation],
+    ],
+    ["video", lb_types.VideoData, [video_mask_annotation]],
+]
+
+
+def create_data_row_for_project(project, dataset, data_row_ndjson, batch_name):
+    data_row = dataset.create_data_row(data_row_ndjson)
+
+    project.create_batch(
+        batch_name,
+        [data_row.uid],  # sample of data row objects
+        5,  # priority between 1(Highest) - 5(lowest)
+    )
+    project.data_row_ids.append(data_row.uid)
+
+    return data_row
+
+
+def test_import_data_types_by_global_key(
+    client,
+    configured_project,
+    initial_dataset,
+    rand_gen,
+    data_row_json_by_data_type,
+    annotations_by_data_type,
+    export_v2_test_helpers,
+):
+    project = configured_project
+    project_id = project.uid
+    dataset = initial_dataset
+    data_type_class = ImageData
+    set_project_media_type_from_data_type(project, data_type_class)
+
+    data_row_ndjson = data_row_json_by_data_type["image"]
+    data_row_ndjson["global_key"] = str(uuid.uuid4())
+    data_row = create_data_row_for_project(project, dataset, data_row_ndjson,
+                                           rand_gen(str))
+
+    annotations_ndjson = annotations_by_data_type["image"]
+    annotations_list = [
+        label.annotations
+        for label in NDJsonConverter.deserialize(annotations_ndjson)
+    ]
+    labels = [
+        lb_types.Label(
+            data={'global_key': data_row.global_key},
+            annotations=annotations,
+        ) for annotations in annotations_list
+    ]
+
+    def find_data_row(dr):
+        return dr['data_row']['id'] == data_row.uid
+
+    label_import = lb.LabelImport.create_from_objects(client, project_id,
+                                                      f"test-import-image",
+                                                      labels)
+    label_import.wait_until_done()
+
+    assert label_import.state == AnnotationImportState.FINISHED
+    assert len(label_import.errors) == 0
+
+    result = export_v2_test_helpers.run_project_export_v2_task(project)
+    exported_data = list(filter(find_data_row, result))[0]
+    assert exported_data
+
+    label = exported_data['projects'][project.uid]['labels'][0]
+    annotations = label['annotations']
+    objects = annotations['objects']
+    classifications = annotations['classifications']
+    assert len(objects) + len(classifications) == len(labels)
+
+    data_row.delete()
+
+
+def validate_iso_format(date_string: str):
+    parsed_t = datetime.datetime.fromisoformat(
+        date_string)  # this will blow up if the string is not in iso format
+    assert parsed_t.hour is not None
+    assert parsed_t.minute is not None
+    assert parsed_t.second is not None
+
+
+def to_pascal_case(name: str) -> str:
+    return "".join([word.capitalize() for word in name.split("_")])
+
+
+def set_project_media_type_from_data_type(project, data_type_class):
```

Comment:
These exist in test_data_types already, could we move them to helpers and deduplicate?